### ISSUE TITLE
[Feature Request]: Add support for vuetify

### ISSUE THREAD

**Issue by BennyAlex** [vuegg/vuegg#11]
Vuetify is one of the biggest component framework for vue.
It would be nice to have this components available in vuegg.

**Comment by phiwut** [comment]
I think you could make it work by using scoped css (https://vue-loader.vuejs.org/guide/scoped-css.html).

**Comment by BennyAlex** [comment]
@phiwut but you also have to use different components. You have to impement the elements on vuegg both with vuetify and MDC

**Comment by alxpez** [comment]
@phiwut I'm sorry but I seem to be missing your point here, could you elaborate on your thoughts?

@BennyAlex Yes you would need to create the component definitions in vuegg, and not just that, also it would be necessary to find a way to include the components (templates and styles) [on runtime](https://github.com/vuegg/vuegg/issues/11#issuecomment-381367520).

A side note: During the early days of vuegg (before it was event called vuegg), the whole of it used to be build with vuetify, and the first tests used this _framework_ components. The reason I replaced it in favour of [VMA](https://github.com/stasson/vue-mdc-adapter) was:

_(quote from vuetify [website](https://vuetifyjs.com/en/getting-started/quick-start))_
> In order for your application to work properly, **you must wrap it in a v-app component**. This component is used for dynamically managing your content area and is the mounting point for many components.

This even applies for their current _"A la carte"_ option, where the VApp component is required as well, which leads me to the assumption that any other vuetify component should be placed inside this wrapper in order to apply the appropriate styles.

As you could guess, this is very limiting, and it results in unexpected CSS rules overrides for many other elements, both, in vuegg interface as well as in any other elements outside vuetify.

I'm not sure if there's any workaround at the moment, but it's worth trying. Do not hesitate to open a **PR** if you think you've got a solution, any collaboration is welcomed!

---

(it's worth pointing out that vuetify is not a _component library_, but a _component framework_)

**Comment by BennyAlex** [comment]
Yes, vuetify requires that you have a v-app at top of all other vuetify components, mainly for placing the navbar and dialogs in the correct position.
But you are also able to place the v-app in another component such as "vuetifyIndex.vue" and don't use it when loading "mdcIndex.vue". 

For the vueeg interface styles you need to use scoped css.

**Comment by buzzzzer** [comment]
It seems they want to deprecate v-app
https://github.com/vuetifyjs/vuetify/pull/3836

**Comment by mfcarneiro** [comment]
Using the Quasar-framework will help to improve the "pallet" of the components and cut some hard work instead VMA? Could be an option?

**Comment by hubyhuby** [comment]
Hi there,
Any news on a possible vuetify implementation ?
"vue-mdc-adapter" doesn t seem to be as popular as vuetify and prety staled as of today  :   
https://github.com/stasson/vue-mdc-adapter/graphs/contributors

**Comment by Jogai** [comment]
That repo states it is deprecated

**Comment by alxpez** [comment]
Yeah, indeed it would be great, not just to migrate from `vue-mdc-adapter`, but also to implement a system that would allow to use different component libraries on the fly.

I had some thoughts around it from the beginning and started designing parts of vuegg with that in mind, but unfortunately I haven't been actively developing this project for a while now, I'm planning on fixing some current issues, updating some basics and implementing new functionalities as soon as I get some spare time in my hands, maybe some help and definitely some inspiration.

In the meantime, any ideas around this feature will be more than welcome!

**Comment by Jogai** [comment]
Documentation on adding another framework could help. That way everybody can scratch their own itch.

**Comment by doctormay6** [comment]
From the discussion in #35, it sounds like adding third party libraries would be tough to implement. Since [vue-mdc-adapter](https://github.com/stasson/vue-mdc-adapter) has been deprecated over 2 years, what do you think about migrating Vuegg to use Vuetify by default instead? I think that would draw in a lot more users by using Vuetify even if other 3rd party libraries aren't able to be added.
